workspace/src/common/launch/art_oak_launch/launch/art_oak.launch.py

```diff
@@ -0,0 +1,133 @@
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.#
+
+# ros imports
+from launch import LaunchDescription
+from launch.actions import SetLaunchConfiguration
+from launch.conditions import IfCondition, UnlessCondition
+from launch_ros.actions import ComposableNodeContainer
+
+# internal imports
+from launch_utils import (
+    IncludeLaunchDescriptionWithCondition,
+    GetLaunchArgument,
+    AddLaunchArgument,
+    SetLaunchArgument,
+)
+
+
+def generate_launch_description():
+    ld = LaunchDescription()
+
+    # ----------------
+    # Launch Arguments
+    # ----------------
+
+    AddLaunchArgument(ld, "use_sim", "False")
+    AddLaunchArgument(ld, "use_sim_time", "False")
+    SetLaunchArgument(
+        ld, "use_sim_time", "True", condition=IfCondition(GetLaunchArgument("use_sim"))
+    )
+
+    AddLaunchArgument(ld, "container", "")
+    container_name = AddLaunchArgument(ld, "container_name", "art_container")
+
+    SetLaunchArgument(
+        ld,
+        "disable_art_sensing",
+        "True",
+        condition=IfCondition(GetLaunchArgument("use_sim")),
+    )
+    SetLaunchArgument(
+        ld,
+        "disable_art_vehicle",
+        "True",
+        condition=IfCondition(GetLaunchArgument("use_sim")),
+    )
+    SetLaunchArgument(
+        ld,
+        "disable_art_simulation",
+        "True",
+        condition=UnlessCondition(GetLaunchArgument("use_sim")),
+    )
+    SetLaunchArgument(
+        ld,
+        "disable_ekf_estimation",
+        "True",
+        condition=IfCondition(GetLaunchArgument("use_sim")),
+    )
+    SetLaunchArgument(
+        ld,
+        "disable_particle_filter_estimation",
+        "True",
+        condition=IfCondition(GetLaunchArgument("use_sim")),
+    )
+
+    # -------------
+    # Composability
+    # -------------
+
+    # If composability is desired, all included launch descriptions should attach to this container and use intraprocess communication
+
+    use_composability = IfCondition(AddLaunchArgument(ld, "use_composability", "False"))
```

Comment:
should be in a common launch file right?

Reply:
I'm assuming this is for having multiple containers communicate on the same ros network, right? Idk how this works, so not sure what you are suggesting I do here..

Do you mean that there should be an `art_common_launch` or something, that all vehicles launch?

Reply:
Ya i think there should be a common launch files for components like this. And this line refers to [composability](https://docs.ros.org/en/galactic/Tutorials/Intermediate/Composition.html) which is actually only relevant for C++ ROS nodes.

Reply:
Ok I think I fixed this with `art_shared_launch`. I'm a little confused why this is working though, I would have thought there would be issues with defining arguments before adding the composability, but this seems to work fine...